pkg/cmd/repo/fork/fork.go

```diff
@@ -185,13 +185,21 @@ func forkRun(opts *ForkOptions) error {
 		return fmt.Errorf("failed to fork: %w", err)
 	}
 
+	// If the existing owner attempts to fork a repo that they own no error is returned
+	// from the API.  This explicit check provides better feedback in the form of an error and
+	// stops proceeding to renaming if specified using --fork-name
+	if ghrepo.FullName(repoToFork) == ghrepo.FullName(forkedRepo) {
```

Comment:
Please push this check into `api.ForkRepo()` (since I feel the logic belongs there instead) and do the name comparison using `strings.EqualFold` so that the comparison is case-insensitive

Reply:
Thanks @mislav - I have moved the logic.  I preferred to test the logic and return the error before mapping the full repository object.